test-suite-custom-generators/src/main/java/io/micronaut/sourcegen/custom/visitor/GenerateMyBean3Visitor.java

```diff
@@ -72,6 +68,20 @@ public void visitClass(ClassElement element, VisitorContext context) {
                 ))
                 .build()
             )
+            .addMethod(MethodDef.builder("castPrimitive")
+                .addParameter(ParameterDef.of("value", TypeDef.primitive(Double.TYPE)))
+                .returns(TypeDef.primitive(Float.TYPE))
+                .addStatement(new StatementDef.Return(
+                    new ExpressionDef.Cast(
```

Comment:
Please add a default method to `ExpressionDef` `cast`, so this statement can we written as:
```
        new VariableDef.MethodParameter(
            "value",
            TypeDef.primitive(Double.TYPE)
        ).cast(Float.TYPE).returning()
```

or

```
.build(self, parameters) -> parameters.get(0).cast(Float.TYPE).returning()
```